flickr_download/flick_download.py

```diff
@@ -368,16 +391,35 @@ def main():
                         help='Skip the actual download of the photo')
     parser.add_argument('-j', '--save_json', action='store_true',
                         help='Save photo info like description and tags, one .json file per photo')
+    parser.add_argument('-c', '--cache', type=str, metavar='CACHE',
+                        help='Cache results on provided file')
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="increase output verbosity")
     parser.set_defaults(**_load_defaults())
 
     args = parser.parse_args()
 
+    if args.verbose:
+        logging.getLogger().setLevel(logging.DEBUG)
+
+    cache = None
+    if args.cache:
+        cache = get_cache(args.cache)
+        Flickr.enable_cache(cache)
+
+        def signal_handler(signal, frame):
+            save_cache(args.cache, cache)
+            sys.exit(0)
```

Comment:
Don’t we want to exit with a return code, so that the caller can act accordingly? (Or is my signal handler knowledge just really outdated?)

Reply:
good catch, I'm changing it